File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/GaiaResponse.kt
package gaia.sdk

import gaia.sdk.response.type.Mutation
import gaia.sdk.response.type.Query
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonCreator

abstract class GaiaResponse {

    data class QueryResponse @JsonCreator constructor(
        @JsonProperty("data") val data: Query? = null,
        @JsonProperty("extensions")val extensions: Map<String, Any>? = null,
        @JsonProperty("errors")val errors: List<Error>? = null): GaiaResponse()

    data class MutationResponse @JsonCreator constructor(
        @JsonProperty("data") val data: Mutation? = null,
        @JsonProperty("extensions") val extensions: Map<String, Any>? = null,
        @JsonProperty("errors") val errors: List<Error>? = null): GaiaResponse()

    data class Error @JsonCreator constructor(
        @JsonProperty("message") val message: String,
        @JsonProperty("locations") val locations: Array<Location>?,
        @JsonProperty("path") val path: Array<Any>?
    )

    data class Location @JsonCreator constructor(
        @JsonProperty("line") val line: Int,
        @JsonProperty("column") val column: Int
    )

}

File: gaia-sdk-python/graphql/GaiaResponse.py
from graphql.response.type.Query import Query
from graphql.response.type.Mutation import Mutation
from graphql.response.type.Subscription import Subscription


class QueryResponse:
    def data(self) -> Query:
        return self.data

    def extensions(self) -> dict:
        return self.extensions

    def errors(self) -> list:
        return self.errors

class MutationResponse:
    def data(self) -> Mutation:
        return self.data

    def extensions(self) -> dict:
        return self.extensions

    def errors(self) -> list:
        return self.errors

class SubscriptionResponse:
    def data(self) -> Subscription:
        return self.data

    def extensions(self) -> dict:
        return self.extensions

    def errors(self) -> list:
        return self.errors

